src/dirnode.cpp

```diff
@@ -920,18 +920,36 @@ void RDirNode::logic(float dt) {
     since_last_node_change += dt;
 }
 
-void RDirNode::drawDirName(FXFont& dirfont) const{
+void RDirNode::drawDirName(FXFont& dirfont) {
     if(parent==0) return;
     if(gGourceSettings.hide_dirnames) return;
     if(gGourceSettings.dir_name_depth > 0 && gGourceSettings.dir_name_depth < (depth-1)) return;
 
     if(!gGourceSettings.highlight_dirs && since_last_node_change > 5.0) return;
 
     float alpha = gGourceSettings.highlight_dirs ? 1.0 : std::max(0.0f, 5.0f - since_last_node_change) / 5.0f;
+    dirfont.setAlpha(alpha);
 
-    vec2 label_pos = spline.getLabelPos();
+    vec2 new_label_delta(0.0f, 0.0f);
+    if (parent->getProjectedPos().y > projected_pos.y) {
+        // alignBottom
+        new_label_delta.y = dirfont.getHeight();
+    }
+    if (parent->getProjectedPos().x < projected_pos.x) {
+        // alignRight
+        new_label_delta.x = dirfont.getWidth(path_token);
+    }
 
-    dirfont.setAlpha(alpha);
+    if (label_delta != new_label_delta) {
+        vec2 label_diff = new_label_delta - label_delta;
+        if (glm::length(label_diff) < 1) {
```

Comment:
Will have a look. Generally you should try to use length2() instead of length as it doesn't need to do a square root. Also should be < 1.0f rather than 1, though  probably doesn't make much difference in this case.